metaflow/plugins/argo/argo_events_decorator.py

```diff
@@ -0,0 +1,173 @@
+from metaflow.decorators import FlowDecorator
+from metaflow.exception import MetaflowException
+from metaflow.util import is_stringish
+
+# TODO: Support dynamic parameter mapping through a context object that exposes
+#       flow name and user name similar to parameter context
+
+
+# TODO: This decorator interface can be a top-level decorator to support similar
+#       implementations for Netflix Maestro, AWS Step Functions, Apache Airflow...
+class ArgoEventsDecorator(FlowDecorator):
+    name = "trigger"
+    defaults = {
+        "event": None,
+        "events": [],
+        "options": {},  # TODO: introduce support for options
+    }
+
+    def flow_init(
+        self, flow, graph, environment, flow_datastore, metadata, logger, echo, options
+    ):
+        self.triggers = []
+        if sum(map(bool, (self.attributes["event"], self.attributes["events"]))) > 1:
```

Comment:
I am still very very unclear as to why we want to have both, I don't think the syntax `events=[myevent]` is much worse than `event=myevent`. If you really just want something like `events=myevent`, that can be done too but we have never had this pattern before and I am not sure of the value of it. Am I missing something?

Reply:
it is common for a flow to depend on a singular event. As you can see from the implementation, it is just syntactic sugar and quite a bit helpful for the majority of use cases that we will run into.